packit/base_git.py

```diff
@@ -0,0 +1,112 @@
+import git
+import logging
+from typing import Optional
+
+from rebasehelper.specfile import SpecFile
+from packit.config import Config, PackageConfig
+from packit.exceptions import PackitException
+from packit.local_project import LocalProject
+
+logger = logging.getLogger(__name__)
+
+
+class BaseGitClass:
+    def __init__(self, config: Config, package_config: PackageConfig):
+        self.config = config
+        self.package_config = package_config
+        self._specfile = None
+
+    @property
+    def local_project(self) -> LocalProject:
+        """
+        :return: an instance of LocalProject
+        """
+        raise NotImplementedError
+
+    @property
+    def specfile(self) -> SpecFile:
+        """
+        :return: an instance of SpecFile
+        """
+        raise NotImplementedError
+
+    @property
+    def specfile_path(self) -> Optional[str]:
+        """
+        :return: a path to a Spec file
+        """
+        raise NotImplementedError
+
+    def create_branch(
+        self,
+        local_project: LocalProject,
+        branch_name: str,
+        base: str = "HEAD",
+        setup_tracking: bool = False,
+    ) -> git.Head:
+        """
+        Create a new git branch in dist-git
+
+        :param local_project: LocalProject to create branch
+        :param branch_name: name of the branch to check out and fetch
+        :param base: we base our new branch on this one
+        :param setup_tracking: set up remote tracking
+               (exc will be raised if the branch is not in the remote)
+        :return the branch which was just created
+        """
+        # it's not an error if the branch already exists
+        origin = local_project.git_repo.remote("origin")
+        head = local_project.git_repo.create_head(branch_name, commit=base)
+
+        if setup_tracking:
+            if branch_name in origin.refs:
+                remote_ref = origin.refs[branch_name]
+            else:
+                raise PackitException("Remote origin doesn't have ref %s" % branch_name)
+            # this is important to fedpkg: build can't find the tracking branch otherwise
+            head.set_tracking_branch(remote_ref)
+
+        return head
+
+    def checkout_branch(self, local_project: LocalProject, git_ref: str):
```

Comment:
Same as in the previous method, you can delete the `local_project` parameter and use `self.local_project`.